app/models/manageiq/providers/google/cloud_manager/event_catcher/stream.rb

```diff
@@ -0,0 +1,62 @@
+# Helper class responsible for polling a Google Pubsub topic and retreiving
+# event type messages. Parsing of the message is deferred to the caller of
+# #poll.
+class ManageIQ::Providers::Google::CloudManager::EventCatcher::Stream
+  def initialize(ems)
+    @ems = ems
+  end
+
+  # Start capturing events
+  def start
+    @collecting_events = true
+  end
+
+  # Stop capturing events
+  def stop
+    @collecting_events = false
+  end
+
+  # Poll for events (blocks forever until #stop is called)
+  def each_batch
+    while @collecting_events
+      yield events.map { |e| JSON.parse(e.message['data']) }
```

Comment:
I like this implemention, very idiomatic :+1: 

What happens, if `JSON.parse` chokes on invalid json? You might move this to a method that rescues a `JSON::ParserError`